manifests/process_exporter.pp

```diff
@@ -107,12 +111,19 @@
     default => undef,
   }
 
+  if empty($watched_processes) {
```

Comment:
I think it would be easier to use the following Puppet type:
* https://github.com/voxpupuli/puppet-extlib/blob/master/REFERENCE.md#extlibdefault_content